src/Extension/MemberExtension.php

```diff
@@ -39,4 +46,68 @@ public function getDefaultRegisteredMethod()
     {
         return $this->owner->RegisteredMFAMethods()->byId($this->owner->DefaultRegisteredMethodID);
     }
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName(['DefaultRegisteredMethodID', 'HasSkippedMFARegistration', 'RegisteredMFAMethods']);
+
+        if (!$this->currentUserCanViewMFAConfig()) {
+            return $fields;
+        }
+
+        $fields->addFieldToTab(
+            'Root.Main',
+            $methodListField = RegisteredMFAMethodListField::create(
+                'MFASettings',
+                _t(__CLASS__ . '.MFA_SETTINGS_FIELD_LABEL', 'Multi-factor Authentication Settings (MFA)'),
+                $this->owner
+            )
+        );
+
+        // Administrators can't modify registered method
+        if ($this->currentUserCanEditMFAConfig()) {
+            $methodListField->setReadonly(true);
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Determines whether the logged in user has sufficient permission to see the MFA config for this Member.
+     *
+     * @return bool
+     */
+    public function currentUserCanViewMFAConfig()
+    {
+        return (Permission::check(self::MFA_ADMINISTER_REGISTERED_METHODS)
+            || $this->currentUserCanEditMFAConfig());
+    }
+
+    /**
+     * Determines whether the logged in user has sufficient permission to modify the MFA config for this Member.
+     * Note that this is different from being able to _reset_ the config (which administrators can do).
+     * @return bool
+     */
+    public function currentUserCanEditMFAConfig()
+    {
+        return (Security::getCurrentUser() && Security::getCurrentUser()->ID === $this->owner->ID);
+    }
+
+    /**
+     * Return a map of permission codes to add to the dropdown shown in the Security section of the CMS.
+     * array(
+     *   'VIEW_SITE' => 'View the site',
+     * );
+     */
+    public function providePermissions()
+    {
+        $label = _t(
+            __CLASS__ . '.MFA_PERMISSION_LABEL',
+            'View / Reset MFA Configuration for other Members'
+        );
+
+        return [
+            self::MFA_ADMINISTER_REGISTERED_METHODS => $label,
```

Comment:
This shows up in an "other" heading in the permissions list. Would it make sense to put it under `Roles and access permissions` instead?